src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -0,0 +1,485 @@
+                        var comparer = StringComparer.OrdinalIgnoreCase;
+                        if (comparer.Equals(value, "default"))
+                        {
+                            severity = ReportDiagnostic.Default;
+                        }
+                        else if (comparer.Equals(value, "error"))
+                        {
+                            severity = ReportDiagnostic.Error;
+                        }
+                        else if (comparer.Equals(value, "warn"))
+                        {
+                            severity = ReportDiagnostic.Warn;
+                        }
+                        else if (comparer.Equals(value, "info"))
+                        {
+                            severity = ReportDiagnostic.Info;
+                        }
+                        else if (comparer.Equals(value, "hidden"))
+                        {
+                            severity = ReportDiagnostic.Hidden;
+                        }
+                        else if (comparer.Equals(value, "suppress"))
+                        {
+                            severity = ReportDiagnostic.Suppress;
+                        }
+                        else
+                        {
+                            severity = null;
+                            diagnosticBuilder.Add(Diagnostic.Create(
+                                InvalidAnalyzerConfigSeverityDescriptor,
+                                Location.None,
+                                diagId,
+                                value,
+                                analyzerConfigPath));
+                        }
+
+                        if (severity.HasValue)
+                        {
+                            treeBuilder[diagId] = severity.GetValueOrDefault();
+                        }
+                    }
+                    else
+                    {
+                        analyzerBuilder[key] = value;
+                    }
+                }
+            }
+        }
+
+        private static DiagnosticDescriptor InvalidAnalyzerConfigSeverityDescriptor
+            = new DiagnosticDescriptor(
+                "InvalidSeverityInAnalyzerConfig",
+                CodeAnalysisResources.WRN_InvalidSeverityInAnalyzerConfig_Title,
+                CodeAnalysisResources.WRN_InvalidSeverityInAnalyzerConfig,
+                "AnalyzerConfig",
+                DiagnosticSeverity.Warning,
+                isEnabledByDefault: true);
+
+        public Section GlobalSection { get; }
+
+        /// <summary>
+        /// The directory the editorconfig was contained in, with all directory separators
+        /// replaced with '/'.
+        /// </summary>
+        public string NormalizedDirectory { get; }
+
+        /// <summary>
+        /// The path passed to <see cref="AnalyzerConfig.Parse(string, string)"/> during construction.
+        /// </summary>
+        public string PathToFile { get; }
```

Comment:
Structure of this type feels off. The logic is at the top and the properties at the bottom. Typically we do the opposite.

Reply:
Do we? This pattern isn't followed in CompilationOptions, for example.

Reply:
It's certainly more common to have props up top and logic down